File: JobZone.ASP.NET/Entities/UserSession.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace JobZone.ASP.NET.Entities
{
    /// <summary>
    /// Maps from: vn.hoidanit.JobZone.domain.entity.UserSession
    /// Table: user_sessions
    /// </summary>
    [Table("user_sessions")]
    public class UserSession
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("id")]
        public long Id { get; set; }

        [Required]
        [Column("refresh_token_jti")]
        public string RefreshTokenJti { get; set; } = null!;

        [Column("ip_address")]
        public string? IpAddress { get; set; }

        [Column("user_agent", TypeName = "TEXT")]
        public string? UserAgent { get; set; }

        [Required]
        [Column("created_at")]
        public DateTime CreatedAt { get; set; }

        [Required]
        [Column("last_used_at")]
        public DateTime LastUsedAt { get; set; }

        [Required]
        [Column("expires_at")]
        public DateTime ExpiresAt { get; set; }

        // Navigation: Many-to-One with User
        [Column("user_id")]
        public long UserId { get; set; }
        public virtual User User { get; set; } = null!;
    }
}

File: springrest-sosnu/src/main/java/vn/hoidanit/jobhunter/domain/entity/UserSession.java
package vn.hoidanit.JobZone.domain.entity;

import java.time.Instant;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "user_sessions")
@Getter
@Setter
public class UserSession {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    @JsonIgnore // Quan trọng: tránh vòng lặp vô hạn khi serialize
    private User user;

    @Column(nullable = false, unique = true)
    private String refreshTokenJti; // JTI (ID) của Refresh Token

    private String ipAddress;

    @Column(columnDefinition = "TEXT")
    private String userAgent;

    @Column(nullable = false)
    private Instant createdAt;

    @Column(nullable = false)
    private Instant lastUsedAt; // Cập nhật khi refresh

    @Column(nullable = false)
    private Instant expiresAt; // Thời hạn của refresh token

    @PrePersist
    public void onPrePersist() {
        this.createdAt = Instant.now();
        this.lastUsedAt = Instant.now();
    }

    @PreUpdate
    public void onPreUpdate() {
        this.lastUsedAt = Instant.now();
    }
}